package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -298,8 +301,39 @@ def dist_matrix(self):
         return self.results.dist_matrix
 
     def _conclude(self):
+        # Build the full pairwise distance matrix from stored positions
+        # Calculate and store results
+        pos = np.asarray(
+            self.results._positions, dtype=np.float64
+        )  # (n_frames, n_atoms, n_dim)
+        n = pos.shape[0]
+
+        D = np.zeros((n, n), dtype=np.float64)
+
+        metric = self._metric
+        cutoff = self._cutoff
+        weights = self._weights
+
+        for i in range(n):
+            pi = pos[i]
+            for j in range(i + 1, n):
+                pj = pos[j]
+                d = metric(pi, pj, weights=weights)
+                if d <= cutoff:
+                    d = 0.0
+                D[i, j] = d
+                D[j, i] = d
```

Comment:
perhaps if we're initializing with zeros, this is slightly more logical

```suggestion
                if d > cutoff:
                    D[i, j] = d
                    D[j, i] = d
```

Reply:
True would be more reasonable, will commit it quickly